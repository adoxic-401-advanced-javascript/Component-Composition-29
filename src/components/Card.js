import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ children }) => {
  return (
    <>
      {children.map(({ _id, name, image, gifLink }) => (
        <div key={_id}>
          {name && <header><h3>{name}</h3></header>}
          {image && name && <figure>
            <img src={image} alt={name} />
          </figure>}
          {gifLink && <figure>
            <img src={gifLink} />
          </figure>}
          {name && _id && <p>{name} has {_id}</p>}
        </div>
      ))}
    </>
  );
};

Card.propTypes = {
  children: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string,
    image: PropTypes.string,
    gitLink: PropTypes.string
  })).isRequired
};

export default Card;
