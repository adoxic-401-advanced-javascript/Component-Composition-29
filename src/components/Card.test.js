import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card component', () => {
  it('renders Card', () => {
    const children = [{ _id: 'some stuff', name: 'string', image: 'string',  }];
    
    const wrapper = shallow(
      <Card> 
        {children}
      </Card>);
    expect(wrapper).toMatchSnapshot();
  });
});
