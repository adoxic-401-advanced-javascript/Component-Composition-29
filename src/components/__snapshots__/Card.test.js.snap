// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card 1`] = `
<Fragment>
  <div
    key="some stuff"
  >
    <header>
      <h3>
        string
      </h3>
    </header>
    <figure>
      <img
        alt="string"
        src="string"
      />
    </figure>
    <p>
      string
       has 
      some stuff
    </p>
  </div>
</Fragment>
`;
